Documentation/interact_v3.md

```diff
@@ -0,0 +1,182 @@
+# Interact with etcd
+
+Users mostly interact with etcd by putting or getting the value of a key. This section describes how to do that by using etcdctl, a command line tool for interacting with etcd server. The concepts described here should apply to the gRPC APIs or client library APIs.
+
+By default, etcdctl talks to etcd server with v2 API for backward compatibility reason. To make etcdctl talks to etcd with v3 API, we first need to set API version to 3 by setting environment. 
+
+``` bash
+export ETCDCTL_API=3
+```
+
+## Write a key
+
+Applications can store keys into etcd cluster. The stored key will be replicated to all etcd members in the cluster through raft to achieve consistency and reliability.
+
+Here is the command to set the value of key `foo` to `bar`:
+
+``` bash
+$ etcdctl put foo bar
+OK
+```
+
+## Read keys
+
+Applications can read values of keys from etcd cluster. Both range query and single key query are supported. 
+
+Suppose we have keys already stored in etcd as blow:
+
+```
+foo = bar
+foo1 = bar1
+foo3 = bar3
+```
+
+Here is the command to read the value of key `foo`:
+
+```bash
+$ etcdctl get foo
+foo
+bar
+```
+
+Here is the command to range over the keys from `foo` to `foo9`:
+
+```bash
+$ etcdctl get foo foo9
+foo
+bar
+foo1
+bar1
+foo3
+bar3
+```
+
+## Read past version of keys
+
+Applications might want to read past version of a key for various reasons. For example, applications might want to rollback to an old configuration by getting an early version of a key or applications might want to get a consistent view over multiple keys over multiple requests. Every modification of the key changes the global revision of etcd cluster. To read an old version of key, application can specify a historical revision.
```

Comment:
Applications may want to read superseded versions of a key for various reasons. For example,  an application may wish to rollback to an old configuration by accessing an early version of a key. Alternatively, an application may want a consistent view over multiple keys over multiple requests by accessing key history.
Since every modification to the etcd cluster key-value store increments the global revision of etcd cluster, an application can read superseded keys by providing a historical etcd revision.